### PR TITLE
🐛Syncer namespace cleanup refactor

### PR DESCRIPTION
**Pull request by jmprusi**
## Summary

Refactor how the syncer handles the deletion of downstream namespaces.

* Removes the upstream namespace controller from syncer.
* Fixes an incorrect use of indexes. 
* Surface the syncer `--downstream-namespace-clean-delay` flag and set it to 2s for tests.
* Fixes an incorrect annotation removal on a test.

### CODE REVIEW

**Review by jmprusi — COMMENTED** [review]
-

File: pkg/syncer/resourcesync/controller.go [hunk 1]
```diff
@@ -466,3 +463,11 @@ func getAllGVRs(synctarget *workloadv1alpha1.SyncTarget) map[schema.GroupVersion
 
 	return gvrs
 }
+
+func IndexByNamespace(obj interface{}) ([]string, error) {
```
Comment (stevekuznetsov) [inline pkg/syncer/resourcesync/controller.go:467]: Can we please use the upstream index.

Comment (jmprusi) [reply pkg/syncer/resourcesync/controller.go]: thanks! done!

**Review by davidfestal — APPROVED** [review]

**Pull request merged**